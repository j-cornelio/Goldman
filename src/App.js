import React            from 'react';
import RaisedButton     from 'material-ui/RaisedButton';
import FlatButton       from 'material-ui/FlatButton';
import questions        from './data/questions';

class App extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      finished    : false,
      stepIndex   : 0,

      questions,
      score: 0
    };
  }

  handleNext(){
    const {stepIndex} = this.state;
    this.setState({
      stepIndex: stepIndex + 1,
      finished: stepIndex >= 3,
    });
  };

  handlePrev(){
    const { stepIndex } = this.state;
    if (stepIndex > 0) {
      this.setState({stepIndex: stepIndex - 1});
    }
  };

  _respond(response, idx){
    var { score } = this.state;
    if(this.state.questions[idx].answer.text === response){
      this.setState({score:  score += this.state.questions[idx].answer.points})
    }
  }

  _renderSteps(index){
    const { questions } = this.state;
    return (
      <div>
        <h2>{questions[index].header}</h2>
        <p>{questions[index].subhead}</p>
        {questions[index].quest.map( (item, i) => {
            return <button className="questionsButtons" onClick={this._respond.bind(this, item, i)} key={i}>{item}</button>
        })}
      </div>
    )
  }

  _getStepContent(stepIndex) {
    var element = null;
    this.state.questions.forEach( (item, idx) => {
      if(stepIndex === idx){
        element = this._renderSteps(stepIndex)
      }
    })

    return element;
  }

  render() {
    const { finished, stepIndex } = this.state,
          contentStyle            = {margin: '0 16px'};

    return (
      <div style={{width: '100%', maxWidth: 700, margin: 'auto'}}>

          { (stepIndex < 4) ? <p id="steps">Step {stepIndex + 1} of 4</p> : null}

          <div style={contentStyle}>
            {finished ? (
              <div>
                <h1>Your Total Score: {this.state.score}</h1>

                { this.state.score > 50 ? <h1>CONGRATULATIONS</h1> : <p>please try again</p> }

                <a
                  href="#"
                  onClick={(event) => {
                    event.preventDefault();
                    this.setState({stepIndex: 0, finished: false, score: 0});
                  }}
                >
                  Reset test
                </a>
              </div>
            ) : (
              <div>
                <div style={{marginTop: 12}}>
                  <span onClick={this.handlePrev.bind(this)} id="prev"></span>

                  <span onClick={this.handleNext.bind(this)} id="forward"></span>
                </div>
                <div>{this._getStepContent(stepIndex)}</div>
              </div>
            )}
          </div>
      </div>
    );
  }
}
export default App;