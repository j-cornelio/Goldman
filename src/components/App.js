import React from 'react';
// Link component is just anchor to match it's path and Route path to load JSX UI Component in prop.children


const App = (props) => {
  return (
    <div className="container">
      <nav className="navbar navbar-default">
        <div className="container-fluid">
          <div className="navbar-header">
            <a className="navbar-brand" href="#">Tutorial part 2</a>
          </div>
          <div className="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
            <ul className="nav navbar-nav">
              <li><Link to="/">Home</Link></li>
            </ul>
          </div>
        </div>
      </nav>
      {/* load components like composition */}
    </div>
  )
}

export default App;

