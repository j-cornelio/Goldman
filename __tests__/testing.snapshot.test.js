import App 		from '../src/App';
import renderer 	from 'react-test-renderer';
import React 		from 'react';

test('Todo component renders the todo correctly', () => {
  const rendered = renderer.create(
    <App  />
  );
  expect(rendered.toJSON()).toMatchSnapshot();
});
