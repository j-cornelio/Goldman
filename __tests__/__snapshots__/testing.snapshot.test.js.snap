// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component renders the todo correctly 1`] = `
<div
  style={
    Object {
      "margin": "auto",
      "maxWidth": 700,
      "width": "100%",
    }
  }
>
  <p
    id="steps"
  >
    Step 
    1
     of 4
  </p>
  <div
    style={
      Object {
        "margin": "0 16px",
      }
    }
  >
    <div>
      <div
        style={
          Object {
            "marginTop": 12,
          }
        }
      >
        <span
          id="prev"
          onClick={[Function]}
        />
        <span
          id="forward"
          onClick={[Function]}
        />
      </div>
      <div>
        <div>
          <h2>
            What is JavaScript?
          </h2>
          <p>
            JavaScript simply is dummy Lorem Ipsum simply is dummy Lorem Ipsum simply is dummy
          </p>
          <button
            className="questionsButtons"
            onClick={[Function]}
          >
            class based
          </button>
          <button
            className="questionsButtons"
            onClick={[Function]}
          >
            Object oriented and dynamic
          </button>
          <button
            className="questionsButtons"
            onClick={[Function]}
          >
            static
          </button>
          <button
            className="questionsButtons"
            onClick={[Function]}
          >
            complied program
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
